### PR TITLE
Comparison of sagas

### PR DESCRIPTION
**Pull request by Zatorg**
Bachelorarbeit Maximilian Krischan

### CODE REVIEW

File: runs/comparisons/suprb_all_tuning.py [hunk 1]
```diff
@@ -227,7 +227,7 @@ def suprb_space(trial: Trial, params: Bunch):
             params.solution_composition__crossover = getattr(suprb.optimizer.solution.ga.crossover, params.solution_composition__crossover)()  # nopep8
 
             params.solution_composition__crossover__crossover_rate = trial.suggest_float('solution_composition__crossover__crossover_rate', 0.7, 1.0)  # nopep8
-            if isinstance(params.solution_composition__crossover__crossover_rate, suprb.optimizer.solution.ga.crossover.NPoint):
```
Comment (RomanSraj) [inline runs/comparisons/suprb_all_tuning.py:230]: Why?

File: runs/saga_comparison_tests/saga_comparison_tuning.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+        n_jobs=4,
+        n_calls=1000,
+        timeout=60*60*72,  # 72 hours
+        scoring='neg_mean_squared_error',
+        verbose=10
+    )
+    
+
+    @param_space()
+    def suprb_space(trial: Trial, params: Bunch):
```
Comment (RomanSraj) [inline runs/saga_comparison_tests/saga_comparison_tuning.py:79]: Move parameters that are the same out of the `if/elif`-structure. That way it becomes better readable and the file doesn't become so big

**Review by RomanSraj — CHANGES_REQUESTED** [review]
I just left 2 (minor) comments. Although before merging I would consider removing most (or all?) added sbatch files and add in readme how the sbatch file should look like (if it is too different from already available sbatch files).

And most importantly: Please rebase your branch to the current master and resolve the conflicts. I did a quick rebase and took all of your changes (because I didn't want to accidently remove something you added), but I can still see a lot of changes that have (probably) nothing to do with your commit. That most likely comes from another commit that got merged to master in between. 
In the end your PR should only have the relevant changes in it

**Comment by heidmic** [comment]
Are all requested changes integrated?

**Comment by Zatorg** [comment]
Added some final small changes to make it ready for merge. Could not get more stuff out of the if/elif in the comparison experiment, as the parameters are only shared between some of the compared sagas. Left one example sbatch for the experiment.

**Review by heidmic — APPROVED** [review]

**Review by RomanSraj — APPROVED** [review]

**Pull request merged**